app/dashboard/src/App.tsx

```diff
@@ -278,6 +282,24 @@ function AppRouter(props: AppRouterProps) {
     [projectManagerInstance],
   )
 
+  React.useEffect(() => {
```

Comment:
Prefer please `useSyncExternalStore` : https://react.dev/reference/react/useSyncExternalStore#reference

Reply:
i don't know if it's appropriate here, `localBackend.rootPath` is not reactive

Reply:
(note that we already use `useSyncExternalStore` elsewhere where we *do* need the reactive updates.)

Reply:
I see we subscribe in the useEffect, it'd be better to use useSyncExternalStore for that case

Reply:
sure, but then the parent component will re-render, no? in which case we'd have to move this `useEffect` to its own child component just to avoid that. not that i'm against that, i just figured it wasn't good practice to create a child component just so we can use `useSyncExternalStore`

it's also worth noting that we have similar subscribe functions in a useEffect for our shortcut handling (and i do believe it is standard practice to subscribe in effects anyway...)

and again i need to note that this is not reactive state we are changing

Reply:
> and I do believe it is standard practice to subscribe in effects anyway...

Not exactly. `useSyncExternalStore` subscribes during the render, `useEffect` - after the render, so useSyncExternalStore will always provide fresh data. useSyncExternalStore supports selection: https://thisweekinreact.com/articles/useSyncExternalStore-the-underrated-react-api, and it's a dedicated API for these situations. 

> It's also worth noting that we have similar subscribe functions in a useEffect

If we're not reading in subscribe (e.g. assigning event listeners), then it's okay to use `useEffect`

In that case, we might get an out-of-sync state when we want to retrieve projects but the path is not updated yet. We should be careful with that. It makes it hard to catch async errors.

In general, I don't like this solution. Why we're not changing the path directly in a callback or during a render?
And I'm double down on moving this logic away from the `App` component. Ideally, App should only create deps and the root tree.

Reply:
> Why we're not changing the path directly in a callback or during a render?

because the path should stay in sync with the value in `localStorage`. idk, we could maybe move this event listener into the backend provider, but i'm not sure that's a complete solution

i guess (theoretically at least) we aren't actually using the root directory anymore because we always provide a directory name to all API calls so we could probably remove this code without repercussions.

but anyway... thoughts? is `useSyncExternalStore` in a nested component the solution here then?

Reply:
If we don't need the root path, let's remove it.

Reply:
see current version of code: <https://github.com/enso-org/enso/pull/10601/files?w=1#diff-1496d10f19e32b61e5063c49393ff603adf03db009ffcfae3c093cb58f7cdc21>
lmk if i should rename the file to something else (e.g. revert but add docs)/improve docs/something else

Reply:
oops replied in wrong thread

Reply:
LGTM